From 88b284f5c079536f5151c3f1dcfc0e692ef26da6 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Thu, 22 Sep 2011 14:55:17 +0200
Subject: [PATCH 2/3] analyze: report startup time in plot mode as well

It now prints something like "Startup finished in 1507ms (kernel) + 850ms (userspace) = 2357ms" below the legend.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 src/systemd-analyze |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/src/systemd-analyze b/src/systemd-analyze
index 649d0e1..d0db984 100755
--- a/src/systemd-analyze
+++ b/src/systemd-analyze
@@ -221,6 +221,18 @@ elif sys.argv[1] == 'plot':
 
         draw_text(context, 0, height-border*2, "Legend: Red = Activating; Pink = Active; Dark Pink = Deactivating", hcenter = 0, vcenter = -1)
 
+        if initrd_time > 0:                                                                                  
+                draw_text(context, 0, height-border*2 + bar_height, "Startup finished in %lums (kernel) + %lums (initrd) + %lums (userspace) = %lums" % ( \
+                        initrd_time/1000, \
+                        (start_time - initrd_time)/1000, \
+                        (finish_time - start_time)/1000, \
+                        finish_time/1000), hcenter = 0, vcenter = -1)
+        else:                                                                                                                                  
+                draw_text(context, 0, height-border*2 + bar_height, "Startup finished in %lums (kernel) + %lums (userspace) = %lums" % ( \
+                        start_time/1000, \
+                        (finish_time - start_time)/1000, \
+                        finish_time/1000), hcenter = 0, vcenter = -1)
+	
         surface.finish()
 elif sys.argv[1] in ("help", "--help", "-h"):
         help()
-- 
1.6.6.1

